### PR TITLE
feat(async-loader): implement custom error template functionality

### PR DESCRIPTION
**Pull request by jb-simon**

### CODE REVIEW

File: packages/AsyncLoader/src/Template.unit.mjs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+};
+
+
+const createElement = (document, html) => {
+    const container = document.createElement('div');
+    container.innerHTML = html;
+    return container;
+};
+
+test.only('replaces content of container with content of a template', async(t) => {
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.unit.mjs]: `only` weg. Überall.

File: packages/AsyncLoader/src/Template.unit.mjs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    const { document, errors } = await setup(true);
+
+    const content = createElement(document,
+        `<div data-content-container>Initial</div>
+        <template data-loading-template>Loading ...</template>`);
+    document.body.appendChild(content);
+
+    const template = new Template(content, '[data-content-container]');
+    const container = document.querySelector('[data-content-container]');
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.unit.mjs:28]: DRY. Zwei mal ein String = String geht in ne Variable.

File: packages/AsyncLoader/src/Template.unit.mjs (hunk 1 above)
Comment (fxstr) [reply packages/AsyncLoader/src/Template.unit.mjs]: `…with content of the provided template`(nicht a – a wäre ein zufälliges template)

File: packages/AsyncLoader/src/Template.unit.mjs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    t.throws(() => {
+        new Template(content, '[data-false-content-container]');
+    }, { message: /Could not find container/ });
+
+    const template = new Template(content, '[data-content-container]');
+
+    t.throws(() => {
+        template.generateContent(['[data-loading-template]'], null, true);
+    }, { message: /'Could not find child element that'/ });
+});
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.unit.mjs:89]: Was mir fehlt, ist ein Test für die replace-Funktion. Dabei gern alle Edge-Cases bedenken: Zu viele Keys vorhanden und zu wenige (also ein Key, der keinen Template-String hat).

File: packages/AsyncLoader/src/Template.unit.mjs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    const container = document.querySelector('[data-content-container]');
+    t.is(container.innerHTML, 'Initial');
+
+    template.generateContent(['[data-second-content-template]', '[data-first-content-template]']);
+    t.is(container.innerHTML, 'Second Content');
+
+    t.is(errors.length, 0);
+});
+
+test.only('replaces content of container with content', async(t) => {
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.unit.mjs]: Unterschied zum ersten Test-Case?

Comment (fxstr) [reply packages/AsyncLoader/src/Template.unit.mjs]: Ah, `setContent` vs. `generateContent`. Gern klar machen.

File: packages/AsyncLoader/src/Template.unit.mjs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    const container = document.querySelector('[data-content-container]');
+    t.is(container.innerHTML, 'Initial');
+
+    template.setContent('New Content');
+    t.is(container.innerHTML, 'New Content');
+
+    t.is(errors.length, 0);
+});
+
+test('throws errors if container and template are not found', async(t) => {
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.unit.mjs:73]: `… or …` statt `and`.

**Review by fxstr — CHANGES_REQUESTED** [review]
Top. Gern immer an DRY denken und den Kommentaren mehr Liebe schenken. Die sind absolut zentral, wenn jemand anderes den Code verstehen will, insb. `param` und `return`.

File: packages/AsyncLoader/src/Template.mjs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+/**
+ * Helper class for AsyncLoader to find template content and display it in a designated container
+ */
+export default class Template {
+
+    #rootElement;
+    #contentContainer;
+
+    /**
+     * @param element
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.mjs:11]: `@param {HTMLElement} element` und ´@param {string} contentContainerSelector`. Die zentrale Information hier ist nicht der Name (den seh ich ja aus dem Constructor), sondern der erwartete Typ (solange wir kein Typescript brauchen, zumindest)

File: packages/AsyncLoader/src/Template.mjs [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    constructor(element, contentContainerSelector) {
+        this.#rootElement = element;
+        this.#contentContainer = this.#getContentContainer(contentContainerSelector);
+    }
+
+    /**
+     * Searches and returns a child element of "rootElement" using a css selector
+     * to be used as the container for the content.
+     * @param selector
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.mjs:22]: Typ

File: packages/AsyncLoader/src/Template.mjs [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+     * Gets a child template that matches selector, replaces its content and displays it
+     *
+     * @param {string[]} selectors                    Array CSS selectors.
+     *                                                Finds the first matching template
+     * @param {Object.<string, string>} replacements  Object of entries that should be replaced
+     *                                                in the template's content. Key is the
+     *                                                variables name which will be surrounded by
+     *                                                two curly braces (key 'message' will look for
+     *                                                '{{message}}' to be replaced)
+     * @param throwIfNotFound                         Specify to throw an error if template is not
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.mjs:69]: Typ

File: packages/AsyncLoader/src/Template.unit.mjs [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        `<div data-content-container>Initial</div>
+        <template data-content-template>{{Hello}} {{message}}</template>`);
+    document.body.appendChild(content);
+
+    const containerSelector = '[data-content-container]';
+    const template = new Template(content, containerSelector);
+    const container = document.querySelector(containerSelector);
+    t.is(container.innerHTML, 'Initial');
+
+    template.generateContent(['[data-content-template]'], {'message': 'World', 'anotherValue': 'thatHasNowhereToGo'});
```
Comment (fxstr) [inline packages/AsyncLoader/src/Template.unit.mjs:89]: Top, genau so.

**Review by fxstr — APPROVED** [review]
